### DISCUSSION TITLE
QUESTION - What event script if I add an empty package?

### DISCUSSION THREAD

**Discussion by Ryan747800B** [pyload/pyload#3978] [Help]
I'm trying to understand what script event I have to use if I only add to queue an empty package. I mean, if I add to queue a package called `TEST`, without adding links or other, log file says `INFO Added package TEST containing 0 links`. If now I want that if I do this then it executes a specific script, what's the correct event-folder where I have to put this script? Because of I only see `package_processed`, `package_finished`, etc., but not `package_added`.

My idea is that if I add an empty package called REBOOT, then it runs a script like this

```
#!/bin/bash

shutdown -r 0
```

**Comment by GammaC0de** [comment]
There is no script for that,
You could create a Hook plugin that handles `links_added` event (see [TransmissionRPC.py](https://github.com/pyload/pyload/blob/stable/module/plugins/hooks/TransmissionRPC.py) for an example)

**Comment by Ryan747800B** [comment]
Thank you, I'll take a look. It doesn't seem a simple thing to do but I'll try.

**Comment by GammaC0de** [comment]
:+1:
